### PR TITLE
[SCAL-83563] dev doc search issue

### PR DESCRIPTION
**Pull request by harsh-ts**
Ticket Id: https://thoughtspot.atlassian.net/browse/SCAL-83563

### CODE REVIEW

File: docs/src/pages/index.tsx [hunk 1]
```diff
@@ -182,12 +184,21 @@ const IndexPage = ({ location }) => {
...
+    const allPageIds = getAllPageIds(navContent);
+
+    const results = useFlexSearch(keyword, index, store)
+        .reduce((acc, cur) => {
+            if (!acc.some((data) => data.pageid === cur.pageid)) {
+                acc.push(cur);
+            }
+            return acc;
+        }, [])
+        .filter((eachFlex: SearchQueryResult) => {
```
[the remark this answers is not in the document]
Comment (harsh-ts) [reply docs/src/pages/index.tsx:196]: I will update it, Thanks for pointing out

File: docs/src/pages/index.tsx [hunk 2]
```diff
@@ -182,8 +184,13 @@ const IndexPage = ({ location }) => {
...
     );
 
+    const allPageIds = getAllPageIds(navContent);
+
     const results = useFlexSearch(keyword, index, store).reduce((acc, cur) => {
-        if (!acc.some((data) => data.pageid === cur.pageid)) {
+        const pageIndex = allPageIds.findIndex(
+            (id: string) => id === cur.pageid,
+        );
+        if (!acc.some((data) => data.pageid === cur.pageid) && pageIndex >= 0) {
```
Comment (jbhanu-thoughtspot) [inline docs/src/pages/index.tsx:193]: You can just do `allPageIds.includes(cur.pageid)` instead of calculating the pageIndex.

Comment (jbhanu-thoughtspot) [inline docs/src/pages/index.tsx:193]: You can just do allPageIds.includes(cur.pageid) instead of calculating the pageIndex.

**Review by jbhanu-thoughtspot — CHANGES_REQUESTED** [review]

**Review by lavishmantri — APPROVED** [review]

File: docs/src/pages/index.tsx (hunk 2 above)
Comment (harsh-ts) [reply docs/src/pages/index.tsx]: Changes done

**Review by jbhanu-thoughtspot — APPROVED** [review]
lgtm

**Pull request merged**